app/models/flag.rb

```diff
@@ -46,6 +46,33 @@ def self.find(id)
     Flag.new(persisted: true).from_json(data)
   end
 
+  def self.find_all(*ids)
+    ids = [ids].flatten
+    return [] if ids.empty?
+
+    datas = redis.mget(ids.map{|id| redis_key(id)})
+
+    if datas.length != ids.length
+      missing_ids = datas.map.with_index {|data, ii| data.nil? ? ids[ii] : nil}.compact
+      raise NotAllItemsFound, "Could not find Flags with IDs #{missing_ids}"
+    end
```

Comment:
what happens if `data` and `ids` are the same length but contain different values?

I know Dante used to convert them to `Set` and then compare them, but you can also just do: `datas.sort == ids.sort`

Reply:
totes! should be `if datas.compact.length != ids.length`